### ISSUE TITLE
Spark Double copy token bug

### ISSUE THREAD

**Issue by jeffwadsworth** [magefree/mage#6097]
Attached image of event.  The copied token should not have the legendary type.
![tokenBeingCopiedBug](https://user-images.githubusercontent.com/1496996/70923435-80647980-1fed-11ea-95c9-cda5aee3b16e.jpg)

**Comment by jeffwadsworth** [comment]
Steps to reproduce:
Soul Foundry -> Any legendary card
Create a token.  This correctly does not have the legendary type
Cast Eyes in the Skies to Populate the token.
The token has the legendary type

**Comment by JayDi85** [comment]
It's uses outdated `CreateTokenCopyTargetEffect`. You can search code for `.getCopyFrom()` and found newest copy implementation with working "copy of copies".

```
// save original copy link (handle copy of copies too)
newBluePrint.setCopy(true, (copyFromPermanent.getCopyFrom() != null ? copyFromPermanent.getCopyFrom() : copyFromPermanent));
```

**Comment by jeffwadsworth** [comment]
It is fine until it gets to CardUtil.copyTo(token).from(copyFrom) in CreateTokenCopyTargetEffect().

**Comment by jeffwadsworth** [comment]
If you see the issue, I will leave it alone.

**Comment by JayDi85** [comment]
Must be fixed via 23ef0e4269944cce2300b7247ce4d2bf68857602
